hathor/reactor/reactor.py

```diff
@@ -14,18 +14,31 @@
 
 from typing import cast
 
-from twisted.internet import reactor as twisted_reactor
 from twisted.internet.interfaces import IReactorCore, IReactorTCP, IReactorTime
 from zope.interface.verify import verifyObject
 
 from hathor.reactor.reactor_protocol import ReactorProtocol
 
-assert verifyObject(IReactorTime, twisted_reactor) is True
-assert verifyObject(IReactorCore, twisted_reactor) is True
-assert verifyObject(IReactorTCP, twisted_reactor) is True
+# Internal variable that should NOT be accessed directly.
+_reactor: ReactorProtocol | None = None
 
-"""
-This variable is the global reactor that should be imported to use the Twisted reactor.
-It's cast to ReactorProtocol, our own type that stubs the necessary Twisted zope interfaces, to aid typing.
-"""
-reactor = cast(ReactorProtocol, twisted_reactor)
+
+def get_global_reactor() -> ReactorProtocol:
```

Comment:
Maybe add a `initialize: bool = False` named attribute so we can avoid wrong initializations. Only the first call will set this to `True`. All other calls must set it to `False`.

Reply:
This is changed in the next PR: https://github.com/HathorNetwork/hathor-core/pull/889#discussion_r1427183991

Reply:
Great! Thanks!